core/api/src/services/ledger/schema.ts

```diff
@@ -172,13 +172,13 @@ const transactionMetadataSchema = new Schema<TransactionMetadataRecord>(
       type: String,
       index: true,
     },
+    external_id: String,
   },
   { id: false },
 )
 
-transactionMetadataSchema.index({
-  hash: 1,
-})
+transactionMetadataSchema.index({ hash: 1 })
+transactionMetadataSchema.index({ external_id: 1 })
```

Comment:
Index needs to enforce uniqueness

Reply:
Field lives in `transactions` and `invoices` collections now, uniqueness enforced at `invoices` persistence level, we won't be able to do the same for transactions since we'd have multiple accounting entries for each external id